common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -39,8 +37,16 @@ public static void cleanDirectBuffer(ByteBuffer buffer) {
       return;
     }
     if (buffer.isDirect()) {
-      Cleaner cleaner = ((DirectBuffer) buffer).cleaner();
-      cleaner.clean();
+      try {
+        final Method getCleanerMethod = buffer.getClass().getMethod("cleaner");
+        getCleanerMethod.setAccessible(true);
+        final Object cleaner = getCleanerMethod.invoke(buffer);
+        if (cleaner != null) {
+          cleaner.getClass().getMethod("clean").invoke(cleaner);
+        }
+      } catch (Exception e) {
```

Comment:
@uronce-cc I was thinking to avoid incompatibility for non-oracle JVMs, but I'm not sure if there are any JVMs that would make direct byte buffer 'GC'ed, so probably you are right. Will throwing an exception make the client & worker code break in non-oracle JVMs?